### PR TITLE
Show container wait error

### PR DESCRIPTION
**Pull request by kolyshkin**
If container wait has failed, show an error from the engine and return an appropriate exit code.

This relies on engine changes from https://github.com/moby/moby/pull/34999

### CODE REVIEW

**Comment by dnephin** [comment]
Looks like we are missing unit test coverage for this function. Would you mind writing a test for the error case and the success case? It should be easy to produce an error using the fake client in `cli/command/container/client_test.go` (just add the wait method), and `NewFakeCli()`.

**Comment by kolyshkin** [comment]
@dnephin test cases added, PTAL

File: cli/command/container/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			exitCode: 125,
+		},
+		{
+			cid:      "non-existent-container-id",
+			exitCode: 125,
+		},
+	}
+
+	client := test.NewFakeCli(&fakeClient{waitFunc: waitFn, Version: api.DefaultVersion})
+	//t.Logf("client version: %v\n", client.ClientVersion())
```
Comment (vdemeester) [inline cli/command/container/utils_test.go]: You can remove that one 👼

File: cli/command/container/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"golang.org/x/net/context"
+)
+
+func waitFn(cid string) (<-chan container.ContainerWaitOKBody, <-chan error) {
+	resC := make(chan container.ContainerWaitOKBody)
+	errC := make(chan error, 1)
+	var res container.ContainerWaitOKBody
+
+	go func() {
+		if strings.Contains(cid, "exit-code-42") {
```
Comment (dnephin) [inline cli/command/container/utils_test.go]: nit: could use a `switch/case`

```go
switch {
case strings.Contains(cid, "exit-code-42"):
 ...
}
```

but not a big deal

**Comment by kolyshkin** [comment]
Commits updated as per @vdemeester and @dnephin suggestions; I think it's ready to be merged.

**Pull request merged**